### ISSUE TITLE
Rocket Jumper rocket cancels Concheror ranked up healing

### ISSUE THREAD

**Issue by joshuadinada** [mastercomfig/team-comtress-2#279]
### Description

As Soldier, if you equip the Concheror and the Rocket Jumper, take damage, wait for the regeneration to rank up, and then shoot a rocket at your feet. The regeneration will now be reset to 1. (This usually happens when you take damage).

### Checklist

- [x] None of the open *or* closed issues document this problem.
- [x] This is a bug, not a new idea.
- [x] This is objectively a bug, and not something that people commonly disagree on.
- [x] This bug is in the code of the game, not in the textures, sounds, or other resources.
- [x] This issue occurs on the latest vanilla TF2 update.
- [ ] This issue is listed on the [Source 1 official bugtracker](https://github.com/ValveSoftware/Source-1-Games/issues).
- [ ] This issue occurs with DirectX level 100.
- [x] This issue occurs on the latest Team Comtress release, or the `master` branch.
- [x] This issue occurs when using no custom content.
- [x] This issue occurs when using `start_tf2.bat` (or `start_server.bat`) with no custom launch options.
- [x] This issue occurs when using a listen server (using the `map` command ingame).
- [ ] This issue occurs when using a dedicated server.
- [ ] This bug is in the internal procedures or tooling (e.g. the build process).

### Reproduction

1. _Already documented in Description_

### Screenshots

Before rocket:
![Immagine 2020-09-20 191130](https://user-images.githubusercontent.com/71554143/93720660-cf49cb80-fb8a-11ea-8441-92dd010eb902.png)
After rocket:
![Immagine 2020-09-20 191315](https://user-images.githubusercontent.com/71554143/93720673-de307e00-fb8a-11ea-9c5e-290cf6e950d4.png)

**Comment by maxwellhupig** [comment]
both the sticky and rocket jumper counts as 0 damage so it makes the crit heals reset and gas passer activate

**Comment by joshuadinada** [comment]
That doesn't answer anything

**Comment by maxwellhupig** [comment]
sorry, i was just saying that it does 0 damage and how it affects other weapons

**Comment by joshuadinada** [comment]
> I don't think this is a bug. The rocket jumper intentionally applies an on damage effect, rather than applying no damage.

Yeah, but still it should be fixed. It's pretty annoying if you're rocket jumping and for no reason the regeneration canceles
